src/modules/kvs/kvstxn.c

```diff
@@ -431,7 +435,7 @@ static int kvstxn_val_data_to_cache (kvstxn_t *kt, int current_epoch,
 }
 
 static int kvstxn_append (kvstxn_t *kt, int current_epoch, json_t *dirent,
-                          json_t *dir, const char *final_name)
+                          json_t *dir, const char *final_name, int *append_count)
```

Comment:
I think your idea of changing this to a flag/boolean is probably a good one that would make its purpose more clear.